src/main/java/uk/gov/hmcts/reform/blobrouter/services/report/ReportService.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.blobrouter.services.report;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.blobrouter.data.ReportRepository;
+import uk.gov.hmcts.reform.blobrouter.model.out.EnvelopeSummaryItem;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+
+import static java.time.ZoneOffset.UTC;
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class ReportService {
+    private final ReportRepository reportRepository;
+
+    public ReportService(ReportRepository reportRepository) {
+        this.reportRepository = reportRepository;
+    }
+
+    public List<EnvelopeSummaryItem> getDailyReport(LocalDate date) {
+
+        var from = date.atStartOfDay().toInstant(UTC);
+        var to = date.atStartOfDay().plusDays(1).toInstant(UTC);
+
+        return reportRepository.getEnvelopeSummary(from, to).stream()
+                .map(s -> new EnvelopeSummaryItem(
+                        s.container,
+                        s.fileName,
+                        toLocalDate(s.fileCreatedAt),
+                        toLocalTime(s.fileCreatedAt),
+                        toLocalDate(s.dispatchedAt),
+                        toLocalTime(s.dispatchedAt),
```

Comment:
We need to handle null values here

Reply:
toLocalDate, toLocalTime return null in case of null value

Reply:
Aaa, sorry, didn't notice those were this class's methods.